### PR TITLE
refreshes syndi-kits and syndicate surplus crates, introduces shared limited stock

### PR DESCRIPTION
**Pull request by flowercuco**
## About The Pull Request

After all, the Syndicate loves a good throwback.

C6O47fPhAB

This PR does a few things with the goal of reimplementing and revitalizing syndicate traitor kits and the syndicate surplus crate. 
Of note is that I have added in a way for limited stock items to share their limited stock. 

Following maintainer guidance the syndicate traitor kits have increased in price and as a result some of the lower value ones have been adjusted. I've given all active bundles current TC costs per item knowing full well they will be inaccurate eventually.

<details>
  <summary>Changes as a result of my audit of syndikits</summary>
    
### UNCHANGED
Recon, Spai, Stealthy, Screwed, Sniper, Nukie Meta, Implants
Mad Scientist, Bees

Lord Singuloth is also unchanged and disabled, I think that it should turn into a new supermatter themed kit maybe. outside of current scope.

### Gun Kit
Replaced emag with doorjack and gave it a chameleon holster, literally moved 1 tc elsewhere

### Murderer
replaced emag again, no additions its a lot of tc and Just Good

### Hacker
added doorjack, otherwise unchanged

### Sabotage
no changes other than adding in extra bombs it didnt have

### James Bond
gave him some gadgets with the freedom implant, emp flashlight, and one x4. also a cyanide pill and deck of cards for fun

### Ninja
Added in miner Jump Boots, smoke spell, and doorjack. dont just want it to be space ninja

### Dark Lord
Added in new lightning bolt spell granter and made the desword default to red. probably overbudget.

### White Whale
dehydrated carp added so you can ride it alongside the ones you grenade out. hard to imagine changing this

### Mr Freeze
changed temperature gun to be cryo only so that i could give him the cryo thermal pistol. cold attacks only.

### 2006 Traitor
doorjack.

</details> 

tl;dr theyre all about 30 tc worth of shit more or less some are more but thats what rarity should be for
you can only buy from one type of syndicrate per round

QOF1WO7CC6

Next up is the return of the surplus crate. 
Crate is generated, gives you gear **based on your progression at the time of buying the crate**, you can use it all at the start and get some chameleon kits and not a lot of dangerous weapons or wait till later. I've changed the weight on some items here and there and given weight to role and species locked items, though I will admit that latter is unimportant because I set moth lanterns to be unable to appear in these two crates. 

dreamseeker_t8abXysKqK

But who cares about that your eyes instantly went to the United Surplus Crate and the United Surplus Key lets be honest.

The united surplus crate is 80 TC worth of uplink items relative to your current progression when you purchase it and gives you a locked box. It **will explode if you try to break it** so be careful with it. It gives you 80 TC and costs 20 TC because it is impossible to open without key. The rub of course is that the Syndicate forbids agents from buying more than one surplus item of any kind, you need to find another traitor and make them buy you a key to open your box. Or I guess you can share the box?

dreamseeker_ts0AZeiyfy

Regardless, if the crate is opened with any other means it doesn't spawn its contents, you need 2 traitor uplinks.
Both of these items have a 30 minute timer because you don't want a crate that has 5 emp flashlights in it. You at least want one energy sword.

I did a lot of code shit and changed various things to be proc based to allow for more editing and interjection of things, as I wrote in code comments making a crate thats locked to a specific set of progression just means changing the proc that generates a list of valid uplink items to check items' progression values to a specified value instead of your characters progression. 

Ok I think that goes over everything more or less????

## Why It's Good For The Game

I've heard that people liked these and I think they are quite fun, being able to go from "i dunno what to do as a traitor" to "ah, of course, I will become the Bombler" is a fun thing to be able to have, and people like to get a bunch of random shit in the mail. Some of it even feels free!!!!!!!!!!!!!!!!!!! Brain points go up!!!

The division of procs allows for more creativity with this system than existed before as well as other possibilities for interacting with the uplink handler in funny ways.

## Changelog

:cl:
add: the syndicate is once again distributing syndi-kits, some now with new technology
add: a fresh batch of syndicate surplus crates have been sent out, though they seem a bit lighter than before
add: in an effort to encourage cooperation, a traitor can now purchase either the new United Surplus Syndicate Crate or its key, but not both
add: lightning bolt book granter for wizard event and one syndie-kit bundle
add: temperature gun that only makes things colder for one syndie-kit bundle
code: it is now possible to have uplink items share limited stock
bal: role-restricted items no longer can be delivered by the stray syndicate drop pod event
/:cl:

### CODE REVIEW

**Comment by Overuseddart** [comment]
Finally although the normal surplus crate seems a bit low, maybe an extra 5tc for each?

File: code/modules/uplink/uplink_items/bundle.dm [hunk 1]
```diff
@@ -51,3 +51,51 @@
...
+	var/static/datum/supply_pack/misc/syndicate/cratefill = new()
+	cratefill.crate_value = starting_crate_value
+	var/obj/structure/closet/crate/surplus_crate = new()
+	cratefill.fill(surplus_crate)
+
+	podspawn(list(
+		"target" = get_turf(user),
+		"style" = STYLE_SYNDICATE,
+		"spawn" = surplus_crate,
+	))
```
Comment (Fikou) [inline code/modules/uplink/uplink_items/bundle.dm:101]: code before subtypes

**Review by Fikou — CHANGES_REQUESTED** [review]
I really like bundles but they certainly feel weird with progression traitor. They were unique gimmicks that you spent your whole TC on, and now you'd still be able to do whatever and even buy multiple.
So, my suggestion: limit their stock to 1 (and make bundle a lower stock of bundle b and vice versa), and raise the price to 25 or so? I think that'll be fair, you're paying for the gimmick items but you aren't forced to use them anymore.

**Comment by EuclidianGeometer** [comment]
If you raise the price to 25, you are requiring folks to do a few extra objectives without any traitor items to get them, which discourages them being used a bit.

**Comment by Jackal-boop** [comment]
Since it's not possible to make a singulo without the SM anymore. you should add an RPD to the singulo bundle

**Comment by Jackal-boop** [comment]
2nd

> being able to purchase a super surplus as a single traitor doesn't sit right with me. the whole risk of it was that you had to meet up with someone and share your tc, could be double-crossed or something.
> 
> of course it's not very likely someone can get to 40 tc without spending any tc but it's plausible which is :/

I got a traitor machine idea cooking in my head and I think it would be able to solve this issue. I'll see if I can code It up

**Comment by Fikou** [comment]
> If you raise the price to 25, you are requiring folks to do a few extra objectives without any traitor items to get them, which discourages them being used a bit.

they are already limited to 30+ minutes. if you stay still for 30 minutes for your bundle that seems lame! either do some risky objectives, or buy stuff and get enough tc to buy the bundle later

**Comment by Overuseddart** [comment]
I like the bundles being implemented back, but in this prs current state the regular surplus crate, as well as the mega one are terrible. 30tc of random items when you can just buy desword and doorjack for more objectives. I’d recommend upping the tc in those crates. Normal 30 -> 35 TC Mega 75 -> 90

**Review by Watermelon914 — CHANGES_REQUESTED** [review]
These all need a limited stock of 1 so that you can't buy multiple.
Also, surplus crate needs to not contain items that exceed a player's progression. Using /datum/supply_pack/misc/syndicate is lazy and I won't accept it. Dynamically build a list of uplink items, there's no issue, the code is not hot at all and you can afford looping through all uplink items to find acceptable items.

I'm willing to accept including items that are at maximum 30% ahead of your current progression.
Additionally, I wouldn't be opposed if you added different tiers of surplus crates that give a chance of unlocking uplink items up to an arbritrary progression value defined for each tier.
For example:
```dm
/datum/uplink_item/bundles_tc/surplus/reputable
	name = "Reputable Surplus Crate"
	desc = "Rumored to contain a valuable assortment of items, but you never know. Contents are sorted to always be worth 40 TC and under 700 reputation"
	cost = 20
	progression_minimum = 30 MINUTES
	starting_crate_value = 75
	/// An item must have a reputation requirement below or equal to this value to be in this crate.
	maximum_item_reputation = 70 MINUTES
```
The reason it is like this is because future items may be added that are locked behind a significantly high progression bar for good reason, and being able to get around this with a surplus crate would be stupid.

**Comment by flowercuco** [comment]
> Also the clip in the PR body makes it seem like the surplus and super surplus can give you goods ahead of the progression curve, like the elite MODsuit. Which defeats the point a bit.

i didnt do my homework and assumed the desword was the limit, my bad!!! ive changed it and the description so that the regular surplus crate now should check your current progression and give you items based on that, and it tells you as such. this has the side effect of slightly changing the random syndie supply pod event to cap out at 40 mins, so that it can still have syndicate bombs mostly. i havent solved for the super surplus crate yet, maybe i will! not be able to easily! ah well

> I really like bundles but they certainly feel weird with progression traitor. They were unique gimmicks that you spent your whole TC on, and now you'd still be able to do whatever and even buy multiple. So, my suggestion: limit their stock to 1 (and make bundle a lower stock of bundle b and vice versa), and raise the price to 25 or so? I think that'll be fair, you're paying for the gimmick items but you aren't forced to use them anymore.

yeah ill raise the price and make sure that none of the bundles are drastically underpriced or whatever, i havent given them limited stock because i dont know how to make their limited stock connected, but i can suss it out this weekened with @Jacquerel 's help

> Since it's not possible to make a singulo without the SM anymore. you should add an RPD to the singulo bundle

done, if you or anyone else has suggestions for the kits let me know, the 2006 one was obvious but i imagine they wont all be.

**Comment by Overuseddart** [comment]
> These all need a limited stock of 1 so that you can't buy multiple. Also, surplus crate needs to not contain items that exceed a player's progression. Using /datum/supply_pack/misc/syndicate is lazy and I won't accept it. Dynamically build a list of uplink items, there's no issue, the code is not hot at all and you can afford looping through all uplink items to find acceptable items.
> 
> I'm willing to accept including items that are at maximum 30% ahead of your current progression. Additionally, I wouldn't be opposed if you added different tiers of surplus crates that give a chance of unlocking uplink items up to an arbritrary progression value defined for each tier. For example:
> 
> ```dm
> /datum/uplink_item/bundles_tc/surplus/super
> 	name = "Reputable Surplus Crate"
> 	desc = "Rumored to contain a valuable assortment of items, but you never know. Contents are sorted to always be worth 40 TC and under 700 reputation"
> 	cost = 20
> 	starting_crate_value = 75
> 	/// An item must have a reputation requirement below or equal to this value to be in this crate.
> 	maximum_item_reputation = 70 MINUTES
> ```
> 
> The reason it is like this is because future items may be added that are locked behind a significantly high progression bar for good reason, and being able to get around this with a surplus crate would be stupid.

The crate is already 300 rep, the only thing ahead would be meme items the desword and the elite mod suit the no slip modules and the other ones no one buys. I think it’s fine to just reduce the weight on those items

**Comment by Watermelon914** [comment]
> > snip
> 
> The crate is already 300 rep, the only thing ahead would be meme items the desword and the elite mod suit the no slip modules and the other mod modules no one buys. I think it’s fine to just reduce the weight on those items, so its rarer to get an elite suit, which is rarely seen for tots. Especially considering that the surplus crates have little tc in them.

I don't care, this is about future proofing, not about what exists now.

**Comment by flowercuco** [comment]
>> snip
> I don't care, this is about future proofing, not about what exists now.

future proofing is another good reason for me to be a bit less lazy than JUST using the crate fill (though, hypothetically, it now checks your reputation) ill see what i can do given that the benefits of not using cratefill include suprlus crate full of clown shit

i did not intend to have these be a way to circumvent prog traitor restrictions so anyone arguing for it being ok for that should stop lol

**Comment by EuclidianGeometer** [comment]
> > If you raise the price to 25, you are requiring folks to do a few extra objectives without any traitor items to get them, which discourages them being used a bit.
> 
> they are already limited to 30+ minutes. if you stay still for 30 minutes not doing anything traitorish as a traitor for your bundle that seems incredibly lame! either do some risky objectives, or buy stuff and get enough tc to buy the bundle later

Yeah you know that's fair enough.

**Comment by YakumoChen** [comment]
> I really like bundles but they certainly feel weird with progression traitor. They were unique gimmicks that you spent your whole TC on, and now you'd still be able to do whatever and even buy multiple. So, my suggestion: limit their stock to 1 (and make bundle a lower stock of bundle b and vice versa), and raise the price to 25 or so? I think that'll be fair, you're paying for the gimmick items but you aren't forced to use them anymore.

If you're raising the price to 25, most of the bundles won't have the value of buying them in a pack, especially since you aren't able to choose which bundle you'll get

**Comment by BlueMemesauce** [comment]
> I like the bundles being implemented back, but in this prs current state the regular surplus crate, as well as the mega one are terrible. 30tc of random items when you can just buy desword and doorjack for more objectives. I’d recommend upping the tc in those crates. Normal 30 -> 35 TC Mega 75 -> 90

yeah 30 is very low, the original surplus crate had 50

**Comment by Overuseddart** [comment]
Maybe this idea is too much hand holding, but maybe can you add a reference to how the surplus crate can only get items %30 above your rep? So players dont buy surplus crates at 300 rep trying to get a desword.

**Comment by flowercuco** [comment]
alright weekend was busy but i fuckin slammed this shit down

few notes that are outside the scope of the new initial post
-FRAME Kit can be used to bypass the need to kill a traitor but then you have to feed it 20 TC AND get it to 300 reputation or wait 30 minutes so I figure thats ok because you still have to kill someone and steal their PDA to do it
-singuloth is disabled, i kept it disabled instead of refreshing it, i think that adding new kits and giving this a new theme would be a separate pr. obviously we're missing one thats a jojo reference.
-did a lot of code proc changes in the handler but i didnt seem to break anything with testing
-auditing/refreshing the syndiekits was very "absolutely no way these things can be 20 TC hell no"
-i raised the payout of the Big Crate, progression is fed to the crates, random purchase shoudl still work, i think that the new Big Crate is really cool

a full list of i've changed is, again, in the first post but i might have missed something. hope not!

**Comment by Overuseddart** [comment]
Was the meta ops bundle changed?

**Comment by flowercuco** [comment]
> Was the meta ops bundle changed?

nope

**Comment by Overuseddart** [comment]
Does bee still give the objectives?

**Comment by flowercuco** [comment]
> Does bee still give the objectives?

you get a little note with the bee saving objective yeah

> (Also does the meta ops still give the hardsuit like how mining on icebox drops some)

nope it was previously changed to a nukie modsuit
recon and screwed continue to give you a syndie space suit thats lightly armoured

File: code/game/objects/items/storage/uplink_kits.dm [hunk 2]
```diff
@@ -61,84 +62,88 @@
...
+			new /obj/item/grenade/syndieminibomb(src) // 6 tc
 
 		if(KIT_IMPLANTS)
-			new /obj/item/implanter/freedom(src)
-			new /obj/item/implanter/uplink/precharged(src)
-			new /obj/item/implanter/emp(src)
-			new /obj/item/implanter/explosive(src)
-			new /obj/item/implanter/storage(src)
+			new /obj/item/implanter/freedom(src) // 5 tc
+			new /obj/item/implanter/uplink/precharged(src) // 4 tc
```
Comment (YakumoChen) [inline code/game/objects/items/storage/uplink_kits.dm:104]: Implant bundle is the "jackpot" bundle, this is a fully TC-charged uplink

File: code/game/objects/items/storage/uplink_kits.dm [hunk 3]
```diff
@@ -61,84 +62,88 @@
...
+			new /obj/item/implanter/emp(src) // 1 tc
+			new /obj/item/implanter/explosive(src) // 2 tc
+			new /obj/item/implanter/storage(src) // 8 tc
+			new /obj/item/implanter/radio/syndicate(src) // 4 tc
```
Comment (YakumoChen) [inline code/game/objects/items/storage/uplink_kits.dm:108]: Gonna be honest it doesn't need the extra implants

File: code/game/objects/items/storage/uplink_kits.dm [hunk 4]
```diff
@@ -183,21 +193,26 @@
 			new /obj/item/storage/belt/chameleon(src) // Unique but worth at least 2 tc
```
Comment (YakumoChen) [inline code/game/objects/items/storage/uplink_kits.dm:193]: The ninja stars are collectively worth whatever the throwing item box in the uplink is. 1tc iirc

File: code/game/objects/items/storage/uplink_kits.dm (hunk 3 above)
Comment (flowercuco) [reply code/game/objects/items/storage/uplink_kits.dm:108]: oh i just misunderstood what this one did. that actually makes it kinda boring to me but youre right it doesnt need the extra ones if it just has 10 TC

**Comment by flowercuco** [comment]
re #72008 
the gun kit obviously should use the new revolver sprite but i think that the 2006 traitor bundle should use the non syndicate branded one. im not like, too precious about this but thats just my gut feeling

**Comment by Imaginos16** [comment]
> re #72008 the gun kit obviously should use the new revolver sprite but i think that the 2006 traitor bundle should use the non syndicate branded one. im not like, too precious about this but thats just my gut feeling

Your choice man.

File: code/game/objects/items/storage/uplink_kits.dm (hunk 3 above)
Comment (YakumoChen) [reply code/game/objects/items/storage/uplink_kits.dm]: the value is that it comes with a whole second uplink worth of TC

File: code/game/objects/items/granters/magic/lightning_bolt.dm [hunk 5]
```diff
@@ -0,0 +1,16 @@
+/obj/item/book/granter/action/spell/lightningbolt
+	granted_action = /datum/action/cooldown/spell/pointed/projectile/lightningbolt
+	action_name = "lightning bolt "
```
Comment (Fikou) [inline code/game/objects/items/granters/magic/lightning_bolt.dm:3]: is that space intended

File: code/game/objects/items/storage/uplink_kits.dm [hunk 6]
```diff
@@ -166,38 +169,48 @@
...
+			new /obj/item/reagent_containers/pill/cyanide(src)
+			new /obj/item/toy/cards/deck/syndicate(src) // 1 tc, for poker
 
 		if(KIT_NINJA)
 			new /obj/item/katana(src) // Unique , hard to tell how much tc this is worth. 8 tc?
 			new /obj/item/reagent_containers/hypospray/medipen/stimulants(src) // 5 tc
 			for(var/i in 1 to 6)
-				new /obj/item/throwing_star(src) // ~5 tc for all 6
+				new /obj/item/throwing_star(src) // 1 tc
 			new /obj/item/storage/belt/chameleon(src) // Unique but worth at least 2 tc
```
Comment (Fikou) [inline code/game/objects/items/storage/uplink_kits.dm:191]: this is a part of the cham kit, its not unique nor is it worth 2 tc really

File: code/game/objects/structures/crates_lockers/crates/syndicrate.dm [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+	max_integrity = 500
+	armor = list(MELEE = 30, BULLET = 50, LASER = 50, ENERGY = 100, BOMB = 0, BIO = 0)
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	integrity_failure = 0 //prevents bust_open from activating
+	/// variable that only lets the crate open if opened by a key from the uplink
+	var/created_items = FALSE
+	/// this is what will spawn when it is opened with a syndicrate key
+	var/list/unlock_contents = list()
+
+/// if the crate takes damage it will explode 24% of the time
```
Comment (Fikou) [inline code/game/objects/structures/crates_lockers/crates/syndicrate.dm:14]: 25

**Comment by Overuseddart** [comment]
Do the syndi crates have unlimited acess to all items or only 30% as suggested, the video shows an elite mod.

**Comment by flowercuco** [comment]
> Do the syndi crates have unlimited acess to all items or only 30% as suggested, the video shows an elite mod.

the united crate is based off of the progression and role of the person who purchases it, i think its more fun to have it be a thing where if you and the other traitor are co-operating, its a choice on who's role is "more valuable" to get the crate with
